source/dub/dub.d

```diff
@@ -105,28 +107,52 @@ class Dub {
 		This property can be altered, so that packages which are downloaded as part
 		of the normal upgrade process are stored in a certain location. This is
 		how the "--local" and "--system" command line switches operate.
-	*/
-	PlacementLocation defaultPlacementLocation = PlacementLocation.user;
 
+		If it is relative, it is resolved relative to the current working directory
+	*/
+	Path defaultRepoPath() @property
+	{
+		return m_defaultRepoPath;
+	}
+	/// Ditto
+	void defaultRepoPath(Path path) @property
+	{
+		path = makeAbsoluteRelativeToCwd(path);
+		//assert(m_packageManager.isManagedPath(path, false), path.toNativeString);
+		m_defaultRepoPath = path;
+	}
+	/// Ditto
+	deprecated defaultPlacementLocation(PlacementLocation location) @property
+	{
+		defaultRepoPath = toPath(location);
+	}
```

Comment:
Currently this property is write-only, this should be the only breaking change in the whole PR. However, if breaking changes aren't possible at this point, an alternative would be to add this:
```D
deprecated PlacementLocation defaultPlacementLocation() @property
{
	if (defaultRepoPath == m_rootPath) return PlacementLocation.local;
    if (defaultRepoPath == m_dirs.userRepo) return PlacementLocation.user;
    if (defaultRepoPath == m_dirs.systemRepo) return PlacementLocation.system;
    throw new Exception("defaultPlacementLocation attempted to be read, "
			~ "but defaultRepoPath is set to a custom path");
}
```